src/org/zaproxy/zap/extension/pscanrulesBeta/CSRFCountermeasures.java

```diff
@@ -216,45 +216,47 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 					evidence = formElement.getFirstElement().getStartTag().toString();
 				}
 				numberOfFormsPassed++;
-			}
-			if (sb.length() > 0) {
-				sb.append(']');
-			}
-			
-			if (!foundCsrfToken && formElements.size() > ignoredForms) {
-				//No known Anti-CSRF tokens found in a form. Not a vulnerability per-se.
-				//but alert it, as a low priority
-				String formDetails = sb.toString();
-				String tokenNamesFlattened = tokenNames.toString();
-				
-				int risk = Alert.RISK_LOW;
-				String desc = Constant.messages.getString("pscanbeta.noanticsrftokens.desc");
-				String extraInfo = Constant.messages.getString("pscanbeta.noanticsrftokens.alert.extrainfo", tokenNamesFlattened, formDetails);
-				if (hasSecurityAnnotation) {
-					risk = Alert.RISK_INFO;
-					extraInfo = Constant.messages.getString("pscanbeta.noanticsrftokens.extrainfo.annotation");
+
+				if (sb.length() > 0) {
+					sb.append(']');
 				}
 				
-			    Alert alert = new Alert(getPluginId(), risk, Alert.CONFIDENCE_MEDIUM,  getName());
-			    alert.setDetail(
-			    			desc + "\n"+getDescription(), 
-				    		msg.getRequestHeader().getURI().toString(),
-				    		"",  //parameter: none.
-				    		"", 
-				    		extraInfo,
-				    		getSolution(), 
-				            getReference(), 
-							evidence,	// Evidence
-							352, // CWE-352: Cross-Site Request Forgery (CSRF)
-							9,	// WASC Id
-				            msg);
-
-			    parent.raiseAlert(id, alert);
+				if (!foundCsrfToken && formElements.size() > ignoredForms) {
+					//No known Anti-CSRF tokens found in a form. Not a vulnerability per-se.
+					//but alert it, as a low priority
+					String formDetails = sb.toString();
+					String tokenNamesFlattened = tokenNames.toString();
+					
+					int risk = Alert.RISK_LOW;
+					String desc = Constant.messages.getString("pscanbeta.noanticsrftokens.desc");
+					String extraInfo = Constant.messages.getString("pscanbeta.noanticsrftokens.alert.extrainfo", tokenNamesFlattened, formDetails);
+					if (hasSecurityAnnotation) {
+						risk = Alert.RISK_INFO;
+						extraInfo = Constant.messages.getString("pscanbeta.noanticsrftokens.extrainfo.annotation");
+					}
+					
+				    Alert alert = new Alert(getPluginId(), risk, Alert.CONFIDENCE_MEDIUM,  getName());
+				    alert.setDetail(
+				    			desc + "\n"+getDescription(), 
+					    		msg.getRequestHeader().getURI().toString(),
+					    		"",  //parameter: none.
+					    		"", 
+					    		extraInfo,
+					    		getSolution(), 
+					            getReference(), 
+								evidence,	// Evidence
+								352, // CWE-352: Cross-Site Request Forgery (CSRF)
+								9,	// WASC Id
+					            msg);
+	
+				    parent.raiseAlert(id, alert);
+				}
+				evidence="";
```

Comment:
(This statement shouldn't be needed.)

Commenting here since I can't add the comments where the variables are being declared:
```Java
StringBuilder sb = new StringBuilder();
String evidence = "";
```
Since the alerts are now being raised per form these should be moved to inside the loop to not share data between them. In the second alert the other info mentions the first form, which is odd:
1. `No known Anti-CSRF tokens [token, csrfToken] were found in the following HTML forms: [Form 1: "zero" ].`
2. `No known Anti-CSRF tokens [token, csrfToken] were found in the following HTML forms: [Form 1: "zero" ]], [Form 2: "name" ].`

The text should also be tweaked to refer to just one form (singular)?